tools/deployment_tools.js

```diff
@@ -0,0 +1,499 @@
+     *                  count (= nonce).
+     */
+    generateAddress() {
+        const addressBytes = Web3.utils.sha3(rlp.encode([this.fromAddress, this.nextNonce]))
+            .slice(12)
+            .substring(14);
+        const address = `0x${addressBytes}`;
+        this.nextNonce += 1;
+
+        return address;
+    }
+}
+
+/**
+ * A contract registry that allows for registering contracts and planning deployment.
+ */
+class ContractRegistry {
+    constructor() {
+        this.contracts = [];
+
+        this.addContract = this.addContract.bind(this);
+        this.toParityGenesisAccounts = this.toParityGenesisAccounts.bind(this);
+    }
+
+    /**
+     * Add a contract to the registry.
+     *
+     * @param {Contract} contracts Contract to add to the registry.
+     */
+    addContract(contract) {
+        this.contracts.push(contract);
+    }
+
+    /**
+     * Add multiple contracts to the registry. See {@link ContractRegistry#addContract}.
+     *
+     * @param {Array.<Contract>} contracts Contracts to add to the registry.
+     */
+    addContracts(contracts) {
+        contracts.forEach(contract => this.addContract(contract));
+    }
+
+    /**
+     * Generate the "accounts" object for a parity chainspec for all the contracts have been added.
+     * For that the { "0x...": { "constructor": "0x..." } } version of account intialization
+     * which is currently exclusive to parity is used.
+     *
+     * See {@link https://wiki.parity.io/Chain-specification} for more details on the
+     * parity chainspec format.
+     *
+     * @param {Object} options.addressGenerator Address generator to use.
+     *                 Defaults to a IncrementingAddressGenerator.
+     *
+     * @returns {object} The "accounts" section for a parity chainspec.
+     */
+    toParityGenesisAccounts(options = {}) {
+        const defaultOptions = {
+            addressGenerator: new IncrementingAddressGenerator(),
+        };
+        const mergedOptions = Object.assign(defaultOptions, options);
+
+        // prepare contracts by ordering and generating addresses
+        const { addressGenerator } = mergedOptions;
+        this._orderContracts();
+        this.contracts.forEach(contract => contract.setAddress(addressGenerator));
+        this.contracts.forEach(contract => contract.instantiate());
+
+        const output = {
+        };
+
+        this.contracts
+            .forEach((contract) => {
+                const address = contract.getAddress();
+                const constructor = contract.constructorData;
+
+                output[address] = {
+                    balance: '0',
+                    constructor,
+                };
+            });
+
+        return output;
+    }
+
+    /**
+     * Generate transaction objects that can be passed as an argument
+     * to `web3.eth.sendTransaction()`.
+     * This allows for deployment on a live network.
```

Comment:
It generates the objects *from this `ContractRegistry`,* right?
Maybe also explain that transactions and the parity genesis account are alternatives.